internal/spokes/spokes.go

```diff
@@ -435,21 +469,30 @@ func (r *SpokesReceivePack) readPack(ctx context.Context, commands []command, ca
 		args...,
 	)
 
+	quarantinePackDir := ""
+	quarantineDir := ""
 	if quarantine := os.Getenv("GIT_SOCKSTAT_VAR_quarantine_dir"); quarantine != "" {
-		packDir := fmt.Sprintf("%s/pack", quarantine)
-		if err := os.MkdirAll(packDir, 0700); err != nil {
+		quarantineDir = quarantine
+		quarantinePackDir = fmt.Sprintf("%s/pack", quarantine)
+		if err := os.MkdirAll(quarantinePackDir, 0700); err != nil {
+			return err
+		}
+	} else {
+		quarantineDir, err = os.MkdirTemp(".", "default-quarantine")
+		if err != nil {
 			return err
 		}
+		quarantinePackDir = fmt.Sprintf("%s/pack", quarantineDir)
+	}
 
-		cmd.Args = append(
-			cmd.Args,
-			filepath.Join(
-				packDir,
-				fmt.Sprintf("quarantine-%d.pack", time.Now().UnixNano()),
-			))
+	cmd.Args = append(
+		cmd.Args,
+		filepath.Join(
+			quarantinePackDir,
+			fmt.Sprintf("quarantine-%d.pack", time.Now().UnixNano()),
```

Comment:
Can this omit the last arg (the output pack name)? The docs for `git index-pack` say it will use "a default name determined from the pack content," which I think is what happens now for incoming pushes, so it will look more normal when the packfile is moved into the repository's object store.

Reply:
The docs also mention that

```
If <pack-file> is not specified, the pack is written to objects/pack/ directory of the current Git repository with a default name determined from the pack content
```
so I think we need to pass the filename because we don't want these file to be created in the `objects/pack` folder? We can try to generate a naming similar to what index-pack is doing if naming is that relevant?

Reply:
@ttaylorr re arguments to index-pack, by default does it create a packfile in `objects/pack/` like the docs say, or does it create it in `$GIT_OBJECT_DIRECTORY/pack/`?